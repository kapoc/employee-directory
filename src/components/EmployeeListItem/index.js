import React from 'react';
import store from '../../store/store';
import { selectEmployee } from '../../actions/actions';

class EmployeeListItem extends React.Component {
  constructor(props){
    super(props);
  }

  selectItem = () => {
    store.dispatch(selectEmployee(this.props.id));
  }

  render(){
    return (
      <>
        {
          this.props.visible ?

          <li className='list-group-item employee-item-wrapper' onClick={this.selectItem}>
            <div className='employee-item'>
              <div className='image-item'>
                <img src={this.props.imgSrc} className='employee-image'/>
              </div>
              <div className='info-item'>
                <span className='name'>
                  {this.props.firstName} {this.props.lastName}
                </span>

                <span className='title'>
                  {this.props.title}
                </span>
              </div>
            </div>

          </li>
          : null
        }
      </>
    );


  }
}

export default EmployeeListItem;
