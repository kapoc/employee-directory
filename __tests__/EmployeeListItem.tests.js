import React from 'react';
import EmployeeListItem from '../src/components/EmployeeListItem/EmployeeListItem';
import renderer from 'react-test-renderer';


describe('EmployeeListItem tests', () => {
  it('renders user list item if it is set to visible', () => {
    const component = renderer.create(
      <EmployeeListItem key={1}
                 visible={true} />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();


    let testInstance = component.root;
    expect(testInstance.findByType('li')).not.toBe(null);
  });
});
