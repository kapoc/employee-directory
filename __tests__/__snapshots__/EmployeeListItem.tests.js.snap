// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmployeeListItem tests renders user list item if it is set to visible 1`] = `
<li
  className="list-group-item employee-item-wrapper"
  onClick={[Function]}
>
  <div
    className="employee-item"
  >
    <div
      className="image-item"
    >
      <img
        className="employee-image"
      />
    </div>
    <div
      className="info-item"
    >
      <span
        className="name"
      >
         
      </span>
      <span
        className="title"
      />
    </div>
  </div>
</li>
`;
